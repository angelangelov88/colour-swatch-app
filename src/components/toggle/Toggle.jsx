import React from 'react';
import { Switch } from '@headlessui/react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

function Toggle(props) {
  const { enabled, label } = props;
  const [active, setActive] = React.useState(false);

  React.useEffect(() => {
    if (enabled) setActive(enabled);
  }, [enabled]);

  return (
    <Switch
      checked={active}
      onChange={setActive}
      className={classNames(
        active ? 'bg-indigo-600' : 'bg-gray-200',
        'relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500',
      )}
    >
      <span className="sr-only">{label}</span>
      <span
        aria-hidden="true"
        className={classNames(
          active ? 'translate-x-5' : 'translate-x-0',
          'pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200',
        )}
      />
    </Switch>
  );
}

Toggle.defaultProps = {
  enabled: false,
  label: '',
};

Toggle.propTypes = {
  enabled: PropTypes.bool,
  label: PropTypes.string,
};

export default Toggle;
export { Toggle };
