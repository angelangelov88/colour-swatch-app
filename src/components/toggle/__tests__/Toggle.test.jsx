import React from 'react';
import renderer from 'react-test-renderer';
import { Toggle } from '../Toggle';

it('should render with default', () => {
  const component = renderer.create(
    <Toggle />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with enabled', () => {
  const component = renderer.create(
    <Toggle enabled />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render not enabled', () => {
  const component = renderer.create(
    <Toggle enabled={false} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with label toggle', () => {
  const component = renderer.create(
    <Toggle label="toggle" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
