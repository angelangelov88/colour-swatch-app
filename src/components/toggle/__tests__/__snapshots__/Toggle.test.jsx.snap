// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render not enabled 1`] = `
<button
  aria-checked={false}
  className="bg-gray-200 relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
  id="headlessui-switch-:r2:"
  onClick={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  role="switch"
  tabIndex={0}
  type="button"
>
  <span
    className="sr-only"
  />
  <span
    aria-hidden="true"
    className="translate-x-0 pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200"
  />
</button>
`;

exports[`should render with default 1`] = `
<button
  aria-checked={false}
  className="bg-gray-200 relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
  id="headlessui-switch-:r0:"
  onClick={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  role="switch"
  tabIndex={0}
  type="button"
>
  <span
    className="sr-only"
  />
  <span
    aria-hidden="true"
    className="translate-x-0 pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200"
  />
</button>
`;

exports[`should render with enabled 1`] = `
<button
  aria-checked={false}
  className="bg-gray-200 relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
  id="headlessui-switch-:r1:"
  onClick={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  role="switch"
  tabIndex={0}
  type="button"
>
  <span
    className="sr-only"
  />
  <span
    aria-hidden="true"
    className="translate-x-0 pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200"
  />
</button>
`;

exports[`should render with label toggle 1`] = `
<button
  aria-checked={false}
  className="bg-gray-200 relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
  id="headlessui-switch-:r3:"
  onClick={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  role="switch"
  tabIndex={0}
  type="button"
>
  <span
    className="sr-only"
  >
    toggle
  </span>
  <span
    aria-hidden="true"
    className="translate-x-0 pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200"
  />
</button>
`;
